packages/vite/src/node/http.ts

```diff
@@ -205,3 +194,34 @@ export function setClientErrorHandler(
     socket.end(`HTTP/1.1 ${msg}\r\n\r\n`)
   })
 }
+
+async function findAvailablePort(
+  port: number,
+  logger: Logger,
+  host?: string,
+  strictPort?: boolean,
+): Promise<number> {
+  return new Promise((resolve, reject) => {
+    if (port > 65535) {
+      reject(new Error('Port not found'))
+    }
```

Comment:
I think it would be nice to have a comment that describes that this code tries to establish a tcp connection instead of trying to listen a port.